### PR TITLE
New dashboard: group migration, showing groups that failed to migrate.

### LINKED ISSUE #1914: [FEATURE]: Add skipping logic for failed groups in the “apply_permissions” task of the "migrate-groups-experimental" workflow
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

Currently the “apply_permissions” task in the "migrate-groups-experimental" workflow fails as soon as we encounter an issue with a group (Shared both the error codes in the additional context). **Can we handle these failure and continue over to the next group while maintaining the list for the failed groups. And persist the failed groups in a new table at the end of the process?** 

**By this we can atleast complete the migration and analyze the problematic groups later. Also, something customer requested**

### Proposed Solution

Skip over to the next group while maintaining the list for the failed groups during the migration. And persist the failed groups in a new table at the end of the process for analysis

### Additional Context

Example error codes:

1. **Preexisting Permissions**:

< 400 Bad Request
< {
<   “error_code”: “BAD_REQUEST”,
<   “message”: “Account group <Group_Name> already has permissions”
< }

2. **Conflicting Permissions**:

< 500 Internal Server Error
< {
<   "error_code": "INTERNAL_ERROR",
<   "message": "Error when updating permissions table"
< }

### PR DESCRIPTION
**Pull request by asnare**
## Changes

This PR introduces a new dashboard focused on group migration. It currently has a widget that displays messages relating to groups that failed to migrate properly as part of the `migrate-groups-experimental` workflow.

### Linked issues

Resolves #1914.

### Functionality

 - New dashboard: `UCX Migration (Groups)`

### Tests

- manually tested
- verified on staging environment (screenshot to be attached)

### CODE REVIEW

**Comment by asnare** [comment]
Screenshots…

The dashboard overview:
image

The dashboard, with no failures detected:
image

The dashboard, with a simulated failure:
image

**Comment by asnare** [comment]
Need to verify that logs are processed (and therefore visible to the dashboard) for failed tasks; the simulation above just logged instead of failing the task.

**Comment by asnare** [comment]
> Need to verify that logs are processed (and therefore visible to the dashboard) for failed tasks; the simulation above just logged instead of failing the task.

This has been verified; logs for failed tasks are also processed.

**Comment by asnare** [comment]
Updated screenshot, showing a failure:
image

File: src/databricks/labs/ucx/queries/migration/groups/00_0_migration_overview.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Workspace Group Migration
+=========================
+
+[Quick link to the group migration documentation.](https://github.com/databrickslabs/ucx/blob/main/docs/local-group-migration.md)
+
+Prerequisites for group migration are:
+
+ - The assessment workflow has completed.
+ - For each workspace group a corresponding account-level group must exist.
```
Comment (FastLee) [inline src/databricks/labs/ucx/queries/migration/groups/00_0_migration_overview.md:13]: Mention the name mapping capabilities, suffix/prefix/id

**Review by FastLee — CHANGES_REQUESTED** [review]

**Review by nfx — APPROVED** [review]
lgtm

File: src/databricks/labs/ucx/queries/migration/groups/00_0_migration_overview.md (hunk 1 above)
Comment (asnare) [reply src/databricks/labs/ucx/queries/migration/groups/00_0_migration_overview.md:13]: @FastLee: In #2344 (commit 56278012f9c958851843c0c9aae61d73b0716e16) I've updated this to mention that there are different strategies, but to keep things brief it's just a reference to the documentation.

**Pull request merged**